language/stdlib/modules/libra_system.move

```diff
@@ -260,14 +216,37 @@ module LibraSystem {
 
     // This function is used in transaction_fee.move to distribute transaction fees among validators
     public fun get_ith_validator_address(i: u64): address {
-        let validator_set = get_validator_set();
-        let len = Vector::length(&validator_set.validators);
-        Transaction::assert(i < len, 3);
-        Vector::borrow(&validator_set.validators, i).addr
+        Vector::borrow(&get_validator_set().validators, i).addr
     }
 
     // Private helper functions
 
+    // The Association, the VM and the validator operator from the current validator set
+    // are authorized to update the set of validator infos
+    fun is_sender_authorized_(): bool {
+        // succeed fast
+        if (Transaction::sender() == 0xA550C18 || Transaction::sender() == 0x0) {
+            return true
+        };
+        let validators = &get_validator_set().validators;
+        // scan the validators to find a match
+        let size = Vector::length(validators);
+        // always true: size > 3 (see remove_validator code)
+
+        let i = 0;
+        if (ValidatorConfig::get_validator_operator_account(Vector::borrow(validators, i).addr) == Transaction::sender()) {
```

Comment:
we should try to break up these lines into something that more conveniently fits on the screen -- aim for a line width of < 100 or < 80 characters.